src/core/gosling.schema.ts

```diff
@@ -242,6 +242,18 @@ export interface GenomicPosition {
     chromosome: string;
     position: number;
 }
+interface OnNewTrackEventData {
+    /** Source visualization ID, i.e., `track.id` */
+    id: string;
+    /** The shape of the source track */
+    shape: BoundingBox | CircularTrackShape;
+}
+interface OnNewViewEventData {
+    /** Source visualization ID, i.e., `track.id` */
+    id: string;
+    /** The shape of the source track */
+    shape: BoundingBox;
+}
```

Comment:
Currently onNewTrack and onNewView simply return the ID and the shape

Reply:
I think we can also add `spec` as what @ThHarbig provided as an example (#926). We can reuse `TrackApiData` and `ViewApiData` for this.

@ThHarbig If you could share with us use cases, that would be helpful for determining what information we want to return.

Reply:
Thinking more about this, I think we can just return an ID of a track or a view when it is mounted and then let users use another API function to get relevant information, such as the bounding box and spec:

```ts
const trackIdOfInterest = 'track-1';
gosRef.current.api.subscribe('onNewTrack', (type, eventData) ⇒ {
   if(eventData.id === trackIdOfInterest) {
      console.log(gosRef.current.api.getTrack('trackIdOfInterest'); // get the bounding box and the spec
   }
});
```

I think this is cleaner as the information we return is not duplicated across API functions.